### PR TITLE
Add contextlib.ExitStack() alternative

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -277,47 +294,29 @@ But this poses a few disadvantages:
...
+            # ...
 
-            with mocker.mock_module.patch.object(SomeClass, 'method', mocked):
-                yield
-        return my_cm
+            stack.enter_context(mock.patch('shutil.copy'))
+            UnixFS.cp('src', 'dst')
+            # ...
 
+But this is arguably a little more complex.
```
Comment (graingert) [inline README.rst]: This is a tricky statement to justify, and it's not clear what is more complex that what.

dym, "Using ``contextlib.ExitStack`` is arguably a little more complex than"...
"pytest-mock" or "using with statements" or "using mock.patch decorators"

Comment (graingert) [reply README.rst]: it's tricky to justify because you could also argue that ``contextlib.ExitStack`` is the simplest of the approaches listed.

Comment (nicoddemus) [reply README.rst]: Agreed, done. 👍

**Comment by nicoddemus** [comment]
Thanks @graingert!

**Pull request merged**